.github/workflows/pytorch_gpu_tests.yml

```
@@ -0,0 +1,40 @@
+name: Pytorch Unit Tests (with GPU) 
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+    paths:
+      - 'ddtrace/profiling/collector/pytorch.py'
+  workflow_dispatch: 
+
+jobs:
+   unit-tests:
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        arch: [x86_64]
+    runs-on: ${{ matrix.os }} 
+    steps:
+      - uses: actions/checkout@v4
+        # Include all history and tags
+        with:
```

```suggestion
        with:
          persist-credentials: false
```

as in [ci: do not persiste git credentials in GitHub Actions](https://github.com/DataDog/dd-trace-py/pull/11640/files)